func.go

```diff
@@ -1072,6 +1073,66 @@ func funcUniqueBy(v, x interface{}) interface{} {
 	return rs
 }
 
+func funcJoinItem(v interface{}) (string, bool) {
+	switch v := v.(type) {
+	case nil:
+		return "", true
+	case string:
+		return v, true
+	case bool:
+		if v {
+			return "true", true
+		} else {
+			return "false", true
+		}
+	case int, float64, *big.Int:
+		return jsonMarshal(v), true
+	default:
+		return "", false
+	}
+}
+
+func funcJoin(v, x interface{}) interface{} {
+	sep, ok := x.(string)
+	if !ok {
+		return &funcTypeError{"join", sep}
+	}
+
+	switch v := v.(type) {
+	case []interface{}:
+		ss := make([]string, len(v))
+		for i, e := range v {
+			if s, ok := funcJoinItem(e); ok {
+				ss[i] = s
+			} else {
+				return &unaryTypeError{name: "join", v: e}
+			}
+		}
+		return strings.Join(ss, sep)
+	case map[string]interface{}:
```

Comment:
The object case got a bit messy. It does the same thing as `.[]`, each in sorted key order which is different from jq.